packages/babel-helper-transform-fixture-test-runner/src/index.js

```diff
@@ -151,7 +156,14 @@ export default function (
               return throwMsg === true || err.message.indexOf(throwMsg) >= 0;
             });
           } else {
-            runTask();
+            if (task.exec.code) {
+              let result = run(task);
+              if (result && typeof result.then === "function") {
+                return result;
```

Comment:
Do we need the `.then` check? I'd have thought we could return it without worrying to let Mocha worry about whether it is a promise.